From 2c9a63e039bc94670304b233ca6f4d4d1eb940c1 Mon Sep 17 00:00:00 2001
From: Andrew Morton <akpm@linux-foundation.org>
Date: Fri, 26 Jul 2019 21:46:39 -0700
Subject: [PATCH 21/21] 
 mm-throttle-allocators-when-failing-reclaim-over-memoryhigh-fix-fix-fix

fix it even more

Reported-by: Nathan Chancellor <natechancellor@gmail.com>
Cc: Chris Down <chris@chrisdown.name>
Cc: Randy Dunlap <rdunlap@infradead.org>
Cc: Stephen Rothwell <sfr@canb.auug.org.au>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/memcontrol.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/memcontrol.c b/mm/memcontrol.c
index 5c7b9facb0eb..224f8413aed6 100644
--- a/mm/memcontrol.c
+++ b/mm/memcontrol.c
@@ -2420,7 +2420,7 @@ void mem_cgroup_handle_over_high(void)
 	clamped_high = max(high, 1UL);
 
 	overage = (u64)(usage - high) << MEMCG_DELAY_PRECISION_SHIFT;
-	do_div(overage, clamped_high);
+	overage /= clamped_high;
 
 	penalty_jiffies = ((u64)overage * overage * HZ)
 		>> (MEMCG_DELAY_PRECISION_SHIFT + MEMCG_DELAY_SCALING_SHIFT);
-- 
2.22.0

